internal/provider/resourcemanager/resource_organization_iam_policy.go

```diff
@@ -64,7 +67,7 @@ func (u *orgIAMPolicyUpdater) GetResourceIamPolicy(ctx context.Context) (*models
 	params.ID = u.client.Config.OrganizationID
 	res, err := u.client.Organization.OrganizationServiceGetIamPolicy(params, nil)
 	if err != nil {
-		diags.AddError("failed to retrieve organization IAM policy", err.Error())
+		diags.Append(customdiags.NewErrorDiagnosticWithErrorCode("failed to retrieve organization IAM policy", err.Error(), status.Code(err)))
```

Comment:
So `status.Code(err)` won't work (99% sure). The issue is that the status.Code() method is expecting the error type to be a grpc/status type: https://github.com/grpc/grpc-go/blob/v1.64.0/status/status.go#L96-L145

Since this error is coming from an HTTP request it is not being typed as a Status error. Instead it will be typed as: https://github.com/hashicorp/hcp-sdk-go/blob/main/clients/cloud-resource-manager/stable/2019-12-10/client/organization_service/organization_service_get_iam_policy_responses.go#L114-L118

The `NewErrorDiagnosticWithErrorCode()` could maybe be replaced to expect a type that has a signature of `GetPayload() *cloud.GoogleRPCStatus` or `Code() int` and use that instead?

Alternatively the error handling here could retrieve and pass the code by converting the error to the underlying type that has the Code() method.